resources/aws_ec2.go

```diff
@@ -684,6 +689,14 @@ func (obj *AwsEc2Res) CheckApply(apply bool) (checkOK bool, err error) {
 		return false, fmt.Errorf("instance is nil")
 	}
 
+	// store useful send/recv values when we return
+	defer func() {
+		// safely dereference the pointers (nil becomes "")
+		obj.PublicIPv4 = aws.StringValue(instance.PublicIpAddress)
+		obj.PrivateIpv4 = aws.StringValue(instance.PublicIpAddress)
```

Comment:
I merged first commit and rebased to master. This line has two bugs in it. Hint, one of them is Ipv4, not IPv4.